shell/platform/android/io/flutter/embedding/engine/plugins/activity/ActivityPluginBinding.java

```diff
@@ -23,6 +25,12 @@
   @NonNull
   Activity getActivity();
 
+  /**
+   * Returns the {@link Lifecycle} associated with the attached {@code Activity}.
+   */
+  @NonNull
+  Lifecycle getLifecycle();
```

Comment:
Lifecycle should be fetched through the flutter_android_lifecycle plugin, and not available directly here